src/bin/sol-fbp-runner/runner.c

```diff
@@ -199,7 +215,7 @@ attach_simulation_nodes(struct runner *r)
         found = false;
 
         for (k = 0; k < ARRAY_SIZE(input_nodes); k++) {
-            if (streq(port_in->packet_type->name, input_nodes[k].packet_name)) {
+            if (streq(port_in->packet_type->name, input_nodes[k].packet_type->name)) {
```

Comment:
You can compare the `packet_types` now instead of making a string comparison. It is cheaper and consistent with what you've said in the commit message.